File: 2404.py
from collections import Counter

class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:
        count = Counter(nums)
        max_freq, min_val = 0, float('inf')

        for num in count.elements():
            if(num % 2 == 0):
                if(count[num] > max_freq):
                    max_freq = count[num]
                    min_val = num
                elif(count[num] == max_freq):
                    min_val = min(min_val, num)

        if(min_val == float('inf')):
            return -1
        return min_val

File: 2404.cpp
class Solution {
public:
    int mostFrequentEven(vector<int>& nums) {
        unordered_map<int, int> count;
        int max_count = -1, max_value = -1;

        for(int i = 0; i < nums.size(); i++) {
            if(nums[i] % 2 == 0) {
                count[nums[i]]++;
                if(count[nums[i]] > max_count) {
                    max_count = count[nums[i]];
                    max_value = nums[i];
                } else if (count[nums[i]] == max_count) {
                    if(nums[i] < max_value) {
                        max_value = nums[i];
                    }
                }
            }
        }
        return max_value;
    }
};